From: Salvo Giangreco <giangrecosalvo9@gmail.com>
Subject: [PATCH] GZD7 right cutout with current inset calculation

Reference sixteen algorithm adaptation; existing width overloads are retained.
GZD7 base resource and static symbols checked; final overlays, DEX and device geometry remain untested.

--- a/smali_classes2/com/android/systemui/dagger/DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl$SwitchingProvider.smali
+++ b/smali_classes2/com/android/systemui/dagger/DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl$SwitchingProvider.smali
@@ -91405,13 +91405,19 @@
 
     move-result-object v4
 
-    iget-object v0, v0, Lcom/android/systemui/dagger/DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl$SwitchingProvider;->referenceSysUIComponentImpl:Lcom/android/systemui/dagger/DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl;
-
-    invoke-static {v0}, Lcom/android/systemui/dagger/DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl;->-$$Nest$mindicatorGardenAlgorithmSidelingCenterCutout(Lcom/android/systemui/dagger/DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl;)Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmSidelingCenterCutout;
-
-    move-result-object v0
-
-    invoke-direct {v1, v2, v3, v4, v0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmFactory;-><init>(Lcom/android/systemui/statusbar/phone/IndicatorCutoutUtil;Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmNoCutout;Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmCenterCutout;Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmSidelingCenterCutout;)V
+    iget-object v5, v0, Lcom/android/systemui/dagger/DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl$SwitchingProvider;->referenceSysUIComponentImpl:Lcom/android/systemui/dagger/DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl;
+
+    invoke-static {v5}, Lcom/android/systemui/dagger/DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl;->-$$Nest$mindicatorGardenAlgorithmSidelingCenterCutout(Lcom/android/systemui/dagger/DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl;)Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmSidelingCenterCutout;
+
+    move-result-object v5
+
+    iget-object v6, v0, Lcom/android/systemui/dagger/DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl$SwitchingProvider;->referenceSysUIComponentImpl:Lcom/android/systemui/dagger/DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl;
+
+    invoke-static {v6}, Lcom/android/systemui/dagger/DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl;->-$$Nest$mindicatorGardenAlgorithmRightCutout(Lcom/android/systemui/dagger/DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl;)Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;
+
+    move-result-object v6
+
+    invoke-direct/range {v1 .. v6}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmFactory;-><init>(Lcom/android/systemui/statusbar/phone/IndicatorCutoutUtil;Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmNoCutout;Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmCenterCutout;Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmSidelingCenterCutout;Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;)V
 
     return-object v1
 
--- a/smali_classes2/com/android/systemui/dagger/DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl.smali
+++ b/smali_classes2/com/android/systemui/dagger/DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl.smali
@@ -25435,6 +25435,16 @@
     .locals 0
 
     invoke-virtual {p0}, Lcom/android/systemui/dagger/DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl;->indicatorGardenAlgorithmNoCutout()Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmNoCutout;
+
+    move-result-object p0
+
+    return-object p0
+.end method
+
+.method public static bridge synthetic -$$Nest$mindicatorGardenAlgorithmRightCutout(Lcom/android/systemui/dagger/DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl;)Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;
+    .locals 0
+
+    invoke-virtual {p0}, Lcom/android/systemui/dagger/DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl;->indicatorGardenAlgorithmRightCutout()Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;
 
     move-result-object p0
 
@@ -35194,6 +35204,28 @@
     return-object v0
 .end method
 
+.method public final indicatorGardenAlgorithmRightCutout()Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;
+    .locals 2
+
+    new-instance v0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;
+
+    iget-object v1, p0, Lcom/android/systemui/dagger/DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl;->referenceGlobalRootComponentImpl:Lcom/android/systemui/dagger/DaggerReferenceGlobalRootComponent$ReferenceGlobalRootComponentImpl;
+
+    iget-object v1, v1, Lcom/android/systemui/dagger/DaggerReferenceGlobalRootComponent$ReferenceGlobalRootComponentImpl;->context:Landroid/content/Context;
+
+    iget-object p0, p0, Lcom/android/systemui/dagger/DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl;->indicatorGardenInputPropertiesProvider:Ldagger/internal/Provider;
+
+    invoke-interface {p0}, Ljavax/inject/Provider;->get()Ljava/lang/Object;
+
+    move-result-object p0
+
+    check-cast p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    invoke-direct {v0, v1, p0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;-><init>(Landroid/content/Context;Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;)V
+
+    return-object v0
+.end method
+
 .method public final indicatorGardenAlgorithmSidelingCenterCutout()Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmSidelingCenterCutout;
     .locals 3
 
--- a/smali_classes3/com/android/systemui/statusbar/phone/IndicatorGardenAlgorithm.smali
+++ b/smali_classes3/com/android/systemui/statusbar/phone/IndicatorGardenAlgorithm.smali
@@ -363,6 +363,53 @@
     return p0
 .end method
 
+.method public final getSideCutoutPadding(II)I
+    .locals 2
+
+    invoke-virtual {p0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->getDefaultSidePadding()I
+
+    move-result v0
+
+    iget-object p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget v1, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->rotation:I
+
+    if-nez v1, :cond_1
+
+    if-lez p1, :cond_0
+
+    iget p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->cutoutSidePaddingD:I
+
+    add-int/2addr p1, p0
+
+    add-int/2addr p2, p0
+
+    invoke-static {p1, p2}, Ljava/lang/Math;->max(II)I
+
+    move-result p0
+
+    invoke-static {p0, v0}, Ljava/lang/Math;->max(II)I
+
+    move-result p0
+
+    goto :goto_0
+
+    :cond_0
+    iget p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->cornerPaddingC:I
+
+    invoke-static {p0, v0}, Ljava/lang/Math;->max(II)I
+
+    move-result p0
+
+    :goto_0
+    return p0
+
+    :cond_1
+    add-int/2addr v0, p2
+
+    return v0
+.end method
+
 .method public hasCameraBottomMargin()Z
     .locals 0
 
--- a/smali_classes3/com/android/systemui/statusbar/phone/IndicatorGardenAlgorithmFactory.smali
+++ b/smali_classes3/com/android/systemui/statusbar/phone/IndicatorGardenAlgorithmFactory.smali
@@ -10,11 +10,13 @@
 
 .field public final indicatorGardenAlgorithmNoCutout:Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmNoCutout;
 
+.field public final indicatorGardenAlgorithmRightCutout:Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;
+
 .field public final indicatorGardenAlgorithmSidelingCenterCutout:Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmSidelingCenterCutout;
 
 
 # direct methods
-.method public constructor <init>(Lcom/android/systemui/statusbar/phone/IndicatorCutoutUtil;Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmNoCutout;Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmCenterCutout;Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmSidelingCenterCutout;)V
+.method public constructor <init>(Lcom/android/systemui/statusbar/phone/IndicatorCutoutUtil;Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmNoCutout;Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmCenterCutout;Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmSidelingCenterCutout;Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;)V
     .locals 0
 
     invoke-direct {p0}, Ljava/lang/Object;-><init>()V
@@ -26,6 +28,8 @@
     iput-object p3, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmFactory;->indicatorGardenAlgorithmCenterCutout:Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmCenterCutout;
 
     iput-object p4, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmFactory;->indicatorGardenAlgorithmSidelingCenterCutout:Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmSidelingCenterCutout;
+
+    iput-object p5, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmFactory;->indicatorGardenAlgorithmRightCutout:Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;
 
     return-void
 .end method
@@ -57,19 +61,28 @@
     return-object p0
 
     :cond_1
+    sget-object v2, Lcom/android/systemui/statusbar/phone/CutoutType;->RIGHT_CUTOUT:Lcom/android/systemui/statusbar/phone/CutoutType;
+
+    if-ne v1, v2, :cond_2
+
+    iget-object p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmFactory;->indicatorGardenAlgorithmRightCutout:Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;
+
+    return-object p0
+
+    :cond_2
     sget-boolean v1, Lcom/android/systemui/BasicRune;->STATUS_LAYOUT_SIDELING_CUTOUT:Z
 
-    if-eqz v1, :cond_2
+    if-eqz v1, :cond_3
 
     invoke-virtual {v0}, Lcom/android/systemui/statusbar/phone/IndicatorCutoutUtil;->isMainDisplay()Z
 
     move-result v0
 
-    if-eqz v0, :cond_2
+    if-eqz v0, :cond_3
 
     return-object v3
 
-    :cond_2
+    :cond_3
     iget-object p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmFactory;->indicatorGardenAlgorithmNoCutout:Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmNoCutout;
 
     return-object p0
--- /dev/null
+++ b/smali_classes3/com/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout.smali
@@ -0,0 +1,303 @@
+.class public final Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;
+.super Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;
+.source "Mesa & Renox, just like bare rocks"
+
+
+# instance fields
+
+.field public final inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+
+# direct methods
+.method public constructor <init>(Landroid/content/Context;Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;)V
+    .locals 0
+
+    invoke-direct {p0, p1, p2}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;-><init>(Landroid/content/Context;Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;)V
+
+    iput-object p2, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    return-void
+.end method
+
+
+# virtual methods
+.method public final calculateCameraTopMargin()I
+    .locals 1
+
+    invoke-virtual {p0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->hasCameraTopMargin()Z
+
+    move-result v0
+
+    if-eqz v0, :cond_0
+
+    iget-object p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->cutoutTopMarginB:I
+
+    goto :goto_0
+
+    :cond_0
+    const/4 p0, 0x0
+
+    :goto_0
+    return p0
+.end method
+
+.method public final calculateCenterContainerMaxWidth()I
+    .locals 1
+
+    invoke-virtual {p0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->getDefaultSidePadding()I
+
+    move-result v0
+
+    mul-int/lit8 v0, v0, 0x2
+
+    iget-object p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->statusBarWidth:I
+
+    sub-int/2addr p0, v0
+
+    int-to-float p0, p0
+
+    const/high16 v0, 0x40400000    # 3.0f
+
+    div-float/2addr p0, v0
+
+    float-to-int p0, p0
+
+    const/4 v0, 0x0
+
+    invoke-static {p0, v0}, Ljava/lang/Math;->max(II)I
+
+    move-result p0
+
+    return p0
+.end method
+
+.method public final calculateLeftContainerMaxWidth(Lcom/android/systemui/statusbar/phone/IndicatorGarden;)I
+    .locals 3
+
+    iget-object v0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget-object v0, v0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->displayCutout:Landroid/view/DisplayCutout;
+
+    const/4 v1, 0x0
+
+    if-eqz v0, :cond_0
+
+    iget-object v0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget-object v0, v0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->displayCutout:Landroid/view/DisplayCutout;
+
+    invoke-virtual {v0}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
+
+    invoke-virtual {v0}, Landroid/view/DisplayCutout;->getSafeInsetLeft()I
+
+    move-result v1
+
+    iget-object v0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget-object v0, v0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->displayCutout:Landroid/view/DisplayCutout;
+
+    invoke-virtual {v0}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
+
+    invoke-virtual {v0}, Landroid/view/DisplayCutout;->getSafeInsetRight()I
+
+    move-result v0
+
+    goto :goto_0
+
+    :cond_0
+    move v0, v1
+
+    :goto_0
+    invoke-virtual {p0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->calculateLeftPadding()I
+
+    move-result v2
+
+    sub-int/2addr v2, v1
+
+    invoke-virtual {p0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->calculateRightPadding()I
+
+    move-result v1
+
+    sub-int/2addr v1, v0
+
+    invoke-virtual {p0, p1, v2, v1}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->getLeftContainerMaxWidth(Lcom/android/systemui/statusbar/phone/IndicatorGarden;II)I
+
+    move-result p0
+
+    return p0
+.end method
+
+.method public final calculateLeftPadding()I
+    .locals 3
+
+    iget-object v0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget-object v1, v0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->displayCutout:Landroid/view/DisplayCutout;
+
+    const/4 v2, 0x0
+
+    if-eqz v1, :cond_0  
+
+    iget-object v0, v0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->displayCutout:Landroid/view/DisplayCutout;
+
+    invoke-virtual {v0}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
+
+    invoke-virtual {v0}, Landroid/view/DisplayCutout;->getSafeInsetLeft()I
+
+    move-result v0
+
+    goto :goto_0
+
+    :cond_0
+    move v0, v2
+
+    :goto_0
+    invoke-virtual {p0, v2, v0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->getSideCutoutPadding(II)I
+
+    move-result p0
+
+    return p0
+.end method
+
+.method public final calculateRightContainerMaxWidth(Lcom/android/systemui/statusbar/phone/IndicatorGarden;)I
+    .locals 3
+
+    iget-object v0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget-object v0, v0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->displayCutout:Landroid/view/DisplayCutout;
+
+    const/4 v1, 0x0
+
+    if-eqz v0, :cond_0
+
+    iget-object v0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget-object v0, v0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->displayCutout:Landroid/view/DisplayCutout;
+
+    invoke-virtual {v0}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
+
+    invoke-virtual {v0}, Landroid/view/DisplayCutout;->getSafeInsetLeft()I
+
+    move-result v1
+
+    iget-object v0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget-object v0, v0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->displayCutout:Landroid/view/DisplayCutout;
+
+    invoke-virtual {v0}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
+
+    invoke-virtual {v0}, Landroid/view/DisplayCutout;->getSafeInsetRight()I
+
+    move-result v0
+
+    goto :goto_0
+
+    :cond_0
+    move v0, v1
+
+    :goto_0
+    invoke-virtual {p0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->calculateLeftPadding()I
+
+    move-result v2
+
+    sub-int/2addr v2, v1
+
+    invoke-virtual {p0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->calculateRightPadding()I
+
+    move-result v1
+
+    sub-int/2addr v1, v0
+
+    invoke-virtual {p0, p1, v2, v1}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->getRightContainerMaxWidth(Lcom/android/systemui/statusbar/phone/IndicatorGarden;II)I
+
+    move-result p0
+
+    return p0
+.end method
+
+.method public final calculateRightPadding()I
+    .locals 4
+
+    iget-object v0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    const/4 v1, 0x0
+    const/4 v2, 0x0
+
+    iget-object v3, v0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->displayCutout:Landroid/view/DisplayCutout;
+
+    if-eqz v3, :padding
+
+    invoke-virtual {v3}, Landroid/view/DisplayCutout;->getSafeInsetRight()I
+    move-result v2
+
+    invoke-virtual {v3}, Landroid/view/DisplayCutout;->getBoundingRectTop()Landroid/graphics/Rect;
+    move-result-object v3
+
+    invoke-virtual {v3}, Landroid/graphics/Rect;->isEmpty()Z
+    move-result v1
+
+    if-nez v1, :no_top_cutout
+
+    iget v3, v3, Landroid/graphics/Rect;->left:I
+    if-lez v3, :no_top_cutout
+
+    iget v0, v0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->statusBarWidth:I
+    sub-int v1, v0, v3
+    if-gez v1, :padding
+
+    :no_top_cutout
+    const/4 v1, 0x0
+
+    :padding
+    invoke-virtual {p0, v1, v2}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->getSideCutoutPadding(II)I
+    move-result p0
+    return p0
+.end method
+
+.method public final hasCameraBottomMargin()Z
+    .locals 1
+
+    const/4 v0, 0x0
+
+    return v0
+.end method
+
+.method public final hasCameraTopMargin()Z
+    .locals 2
+
+    invoke-virtual {p0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->getHasCutoutForIndicator()Z
+
+    move-result v0
+
+    const/4 v1, 0x0
+
+    if-eqz v0, :cond_1
+
+    iget-object p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->rotation:I
+
+    const/4 v0, 0x1
+
+    if-nez p0, :cond_0
+
+    move p0, v0
+
+    goto :goto_0
+
+    :cond_0
+    move p0, v1
+
+    :goto_0
+    if-eqz p0, :cond_1
+
+    move v1, v0
+
+    :cond_1
+    return v1
+.end method
+
